### PR TITLE
[Proposal] 11: Reduce Governance Thresholds

### PR DESCRIPTION
**Pull request by emptysetsquad**
# Proposal 11: Reduce Governance Thresholds

## Summary
- Implements [EIP-8](https://www.emptyset.xyz/t/eip-8-quorum-and-proposal-threshold-reductions/85)
- Several QoL improvements

#### EIP-8
- Lowers the governance proposal threshold from `1%` to `0.5%`.
- Lowers the governance quorum threshold from `33%` to `20%`. 

#### Quality of life improvements
- Removes `Pool 1` migration reward payout.
- Refactors proposal threshold into `Constants.sol`.
- Fully decouples debt repayment from the `increaseSupply` method.
- Adds getters for `lockedUntil` and `fluidUntil` user data.
- Standardizes events for incentivization rewards.

#### Safety
- Adds `3` epoch expiration for approved proposals. This ensures that if governance rules change in the future, we don't accidentally enable an old proposal to be committed.

## Rewards
- Rewards `committer` with `100 ESD`

## Tracking
Status: Pre-proposal

### CODE REVIEW

**Review by Austin-Williams — APPROVED** [review]
```
EIP-8
    Lowers the governance proposal threshold from 1% to 0.5%.
        - ✅ LGTM.
    Lowers the governance quorum threshold from 33% to 20%.
        - ✅ LGTM.

Quality of life improvements
    Removes Pool 1 migration reward payout.
        - ✅ LGTM. No effect on storage layout.
    Refactors proposal threshold into Constants.sol.
        - ✅ LGTM.
    Fully decouples debt repayment from the increaseSupply method.
        - 👀 Concept ACK, but I did not check this carefully (lack of time).
        - 📝 Note: `increaseDebt` now returns a `uint256`. This function is called by `mintToAccount` and `shrinkSupply`. The `mintToAccount` function doesn't catch or use the return value. Nbd.
    Adds getters for lockedUntil and fluidUntil user data.
        - ✅ The new getters look good.
    Standardizes events for incentivization rewards.
        - ✅ LGTM. 

Safety
    Adds 3 epoch expiration for approved proposals. This ensures that if governance rules change in the future, we don't accidentally enable an old proposal to be committed.
        - ✅ LGTM.
        -  ⚠️ Obviously, just make sure nobody ever sets `GOVERNANCE_EXPIRATION = 0` or governance will get bricked.

Other
  - 📝 Note that the commit can (and likely will) be front run. Nbd, it doesn't hurt anything. Just don't be surprised when it happens. :)
```

I would suggest having someone look closely at the refactor of `increaseSupply` to verify its correctness and check that the tests cover all appropriate edge cases there (I didn't have time to do that fully).

**Review by MastaBook — APPROVED** [review]

**Comment by MastaBook** [comment]
Ty

**Review by MastaBook — DISMISSED** [review]

File: protocol/contracts/dao/Comptroller.sol [hunk 1]
```diff
@@ -57,11 +57,13 @@ contract Comptroller is Setters {
...
 
-    function increaseDebt(uint256 amount) internal {
+    function increaseDebt(uint256 amount) internal returns (uint256) {
         incrementTotalDebt(amount);
-        resetDebt(Constants.getDebtRatioCap());
+        uint256 lessDebt = resetDebt(Constants.getDebtRatioCap());
 
         balanceCheck();
+
+        return lessDebt > amount ? 0 : amount.sub(lessDebt);
```
Comment (pranaymohan) [inline protocol/contracts/dao/Comptroller.sol:66]: qq -- when would `lessDebt > amount`? Seems like `amount` should always be greater or equal if debt is below cap initially. And this function should prevent it from ever going over, if I understand correctly.

**Comment by emptysetsquad** [comment]
@Austin-Williams This is actually a good note:

`Obviously, just make sure nobody ever sets GOVERNANCE_EXPIRATION = 0 or governance will get bricked.`

I updated the expiration check to always add at least `1` and for the `GOVERNANCE_EXPIRATION` to represent additional epochs on top of `1`, just to be extra safe.

File: protocol/contracts/dao/Comptroller.sol (hunk 1 above)
Comment (emptysetsquad) [reply protocol/contracts/dao/Comptroller.sol:66]: It's very unlikely, but there could be cases where we've just significantly changed the debt ratio cap or the total supply of `ESD` such that a large amount of the debt is trimmed off here.

This is more of a defensive check since a subtraction underflow here would cause a revert bricking the entire system.

**Review by x230xme — APPROVED** [review]

**Comment by l3wi** [comment]
A contract has been proposed:
http://emptyset.finance/#/governance/candidate/0xdC99214431bEb52877Cde88C81155D29f61D6e09

Diff Checker: https://www.diffchecker.com/pU34rPtG

File: protocol/test/dao/Regulator.test.js [hunk 2]
```diff
@@ -466,6 +467,102 @@ describe('Regulator', function () {
           });
         });
       });
+
```
Comment (jcliff) [inline protocol/test/dao/Regulator.test.js:470]: love the test:code ratio of this patch!
